components/script/dom/webglrenderingcontext.rs

```diff
@@ -471,6 +545,99 @@ impl WebGLRenderingContext {
         return Ok(expected_byte_length);
     }
 
+    fn flip_teximage_y(&self,
+                       pixels: Vec<u8>,
+                       internal_format: TexFormat,
+                       data_type: TexDataType,
+                       width: usize,
+                       height: usize) -> Vec<u8> {
```

Comment:
We're passed in a non-mut Vec at the top of the tex_image()s since we shouldn't be rewriting the JS's array.  We could potentially see that we're going to be chaining more than one of these fixups and use a single mut Vec that we operate on, but that sounded messy.